core/values/consensus.go

```diff
@@ -0,0 +1,56 @@
+package values
+
+// QuorumPhase is a phase that needs quorum to proceed
+type QuorumPhase byte
+
+const (
+	// QuorumPrepare ..
+	QuorumPrepare QuorumPhase = iota
+	// QuorumCommit ..
+	QuorumCommit
+)
+
+// ConsensusMechanism is the way we decide consensus
+type ConsensusMechanism byte
+
+const (
+	// SuperMajorityVote is a 2/3s voting mechanism, pre-PoS
+	SuperMajorityVote ConsensusMechanism = iota
+	// SuperMajorityStake is 2/3s of total staked amount for epoch
+	SuperMajorityStake
+)
+
+// PBFTPhase : different phases of consensus
+type PBFTPhase byte
+
+// Enum for PbftPhase
+const (
+	PBFTAnnounce PBFTPhase = iota
+	PBFTPrepare
+	PBFTCommit
+)
+
+// PBFTState determines whether a node is in normal or viewchanging mode
+type PBFTState byte
+
+// Enum for node Mode
+const (
+	PBFTNormal PBFTState = iota
+	PBFTViewChanging
+	PBFTSyncing
+	PBFTListening
+)
+
+var (
+	modes  = [...]string{"Normal", "ViewChanging", "Syncing", "Listening"}
```

Comment:
Enum values start having undefined holes in their range, so it's usually more appropriate to use a map:
```suggestion
	stateNames  = map[PBFTState]string{
		PBFTNormal:       "Normal",
		PBFTViewChanging: "ViewChanging",
		PBFTSyncing:      "Syncing",
		PBFTListening:    "Listening",
	}
```

Ditto for `phases` below.